cmd/util/ledger/migrations/evm.go

```diff
@@ -0,0 +1,111 @@
+package migrations
+
+import (
+	"fmt"
+
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"github.com/rs/zerolog"
+
+	"github.com/onflow/flow-go/fvm/evm/stdlib"
+	"github.com/onflow/flow-go/fvm/systemcontracts"
+	"github.com/onflow/flow-go/model/flow"
+)
+
+func NewEVMDeploymentMigration(
+	chainID flow.ChainID,
+	logger zerolog.Logger,
+	full bool,
+) RegistersMigration {
+
+	systemContracts := systemcontracts.SystemContractsForChain(chainID)
+	address := systemContracts.EVMContract.Address
+
+	var code []byte
+	if full {
+		code = stdlib.ContractCode(
+			systemContracts.NonFungibleToken.Address,
+			systemContracts.FungibleToken.Address,
+			systemContracts.FlowToken.Address,
+		)
+	} else {
+		code = []byte(stdlib.ContractMinimalCode)
+	}
+
+	return NewDeploymentMigration(
+		chainID,
+		Contract{
+			Name: systemContracts.EVMContract.Name,
+			Code: code,
+		},
+		address,
+		map[flow.Address]struct{}{
+			address: {},
+		},
+		logger,
+	)
+}
+
+// NewEVMSetupMigration returns a migration that sets up the EVM contract account.
+// It performs the same operations as the EVM contract's initializer, calling the function EVM.setupHeartbeat,
+// which in turn creates an EVM.Heartbeat resource, and writes it to the account's storage.
+func NewEVMSetupMigration(
```

Comment:
New migration which creates `EVM.Heartbeat` and saves it to storage.

The resource is created manually, because that is easiest. I tried to use FVM, but it is much more complex and I couldn't get the `InternalEVM` contract to work, see [`daf5c31` (#6251)](https://github.com/onflow/flow-go/pull/6251/commits/daf5c31ffd49621fe66b439d7b14e8d3e5ef6bb5)